File: python-algo/bubble_sort.py
def bubbleSort(unsorted_list:list[int])->list[int]:
    n = len(unsorted_list)

    for i in range (0,n-1): 
        for j in range(0,n-1-i): 
            if unsorted_list[j] > unsorted_list[j+1]: 
                #swap 
                temp = unsorted_list[j+1] 
                unsorted_list[j+1] = unsorted_list[j] 
                unsorted_list[j] = temp 
    return unsorted_list 

def main(): 
    unsorted_list=[3,5,7,4,4,7,2] 
    print(bubbleSort(unsorted_list)) 

main()

File: kotlin-algo/src/bubble_sort.kt
fun bubbleSort(unsortedList: MutableList<Int>): MutableList<Int> {
    val n = unsortedList.size
    var temp: Int
    for (i in 0..<n - 1) {
        for (j in 0..<n - i - 1) {
            if (unsortedList[j] > unsortedList[j + 1]) {
                //swap
                temp = unsortedList[j]
                unsortedList [j] = unsortedList[j + 1]
                unsortedList [j + 1] = temp
            }
        }
    }
    return unsortedList
}

fun main() {
    val unsortedList: MutableList<Int> = mutableListOf(3, 4, -3, 7, 12, 5)
    println (bubbleSort(unsortedList))
}